Command/SystemInfoDumpCommand.php

```diff
@@ -73,36 +73,17 @@ protected function execute(InputInterface $input, OutputInterface $output)
             foreach ($this->registry->getIdentifiers() as $identifier) {
                 $output->writeln("  $identifier", true);
             }
-        } else if ($identifiers = $input->getArgument('info-collectors')) {
-            foreach ($identifiers as $identifier) {
-                $this->outputInfo(
-                    $this->registry->getItem($identifier),
-                    $output
-                );
-            }
+        } else if ($input->getArgument('info-collectors')) {
+            $identifiers = $input->getArgument('info-collectors');
         } else {
-            foreach ($this->registry->getIdentifiers() as $identifier) {
-                $this->outputInfo($this->registry->getItem($identifier), $output);
-            }
+            $identifiers = $this->registry->getIdentifiers();
         }
-    }
-
-    /**
-     * Output info collected by the given collector.
-     *
-     * @param $infoCollector SystemInfoCollector
-     * @param $output OutputInterface
-     */
-    private function outputInfo(SystemInfoCollector $infoCollector, OutputInterface $output)
-    {
-        $infoValue = $infoCollector->collect();
 
-        $outputArray = [];
-        // attributes() is deprecated, and getProperties() is protected. TODO add a toArray() or similar.
-        foreach ($infoValue->attributes() as $property) {
-            $outputArray[$property] = $infoValue->$property;
+        $collectedInfo = [];
+        foreach ($identifiers as $identifier) {
```

Comment:
Don't you get an "undefined variable" warning here when using the `list-info-collectors` option?